ui/app/hooks/useTransactionTimeRemaining.js

```diff
@@ -0,0 +1,92 @@
+import { getEstimatedGasPrices, getEstimatedGasTimes, getFeatureFlags } from '../selectors'
+import { hexWEIToDecGWEI } from '../helpers/utils/conversions.util'
+import { useSelector } from 'react-redux'
+import { useRef, useEffect, useState, useMemo } from 'react'
+import { isEqual } from 'lodash'
+import { getRawTimeEstimateData } from '../helpers/utils/gas-time-estimates.util'
+import { getCurrentLocale } from '../ducks/metamask/metamask'
+
+
+/**
+ * Calculate the number of minutes remaining until the transaction completes.
+ * @param {number} initialTimeEstimate - timestamp for the projected completion time
+ * @param {number} submittedTime       - timestamp of when the tx was submitted
+ * @return {number} minutes remaining
+ */
+function calcTransactionTimeRemaining (initialTimeEstimate, submittedTime) {
+  const currentTime = (new Date()).getTime()
+  const timeElapsedSinceSubmission = (currentTime - submittedTime) / 1000
+  const timeRemainingOnEstimate = initialTimeEstimate - timeElapsedSinceSubmission
+
+  const renderingTimeRemainingEstimate = Math.round(timeRemainingOnEstimate / 60)
+  return renderingTimeRemainingEstimate
+}
+
+/**
+ * returns a string representing the number of minutes predicted for the transaction to be
+ * completed. Only returns this prediction if the transaction is the earliest pending
+ * transaction, and the feature flag for showing timing is enabled.
+ * @param {bool} isPending         - is the transaction currently pending
+ * @param {bool} isEarliestNonce   - is this transaction the earliest nonce in list
+ * @param {number} submittedTime   - the timestamp for when the transaction was submitted
+ * @param {number} currentGasPrice - gas price to use for calculation of time
+ * @returns {string | undefined} i18n formatted string if applicable
+ */
+export function useTransactionTimeRemaining (
+  isPending,
+  isEarliestNonce,
+  submittedTime,
+  currentGasPrice
+) {
+  // the following two selectors return the result of mapping over an array, as such they
+  // will always be new objects and trigger effects. To avoid this, we use isEqual as the
+  // equalityFn to only update when the data is new.
+  const gasPrices = useSelector(getEstimatedGasPrices, isEqual)
+  const estimatedTimes = useSelector(getEstimatedGasTimes, isEqual)
+  const locale = useSelector(getCurrentLocale)
+  const interval = useRef()
+  const [timeRemaining, setTimeRemaining] = useState(null)
+  const featureFlags = useSelector(getFeatureFlags)
+  const transactionTimeFeatureActive = featureFlags?.transactionTime
+
+  const rtf = new Intl.RelativeTimeFormat(locale, { style: 'narrow' })
```

Comment:
It might be better to use `numeric: 'auto'`, so that <1 minute is shown as "this minute" instead of `in 0 min.`

```suggestion
  const rtf = new Intl.RelativeTimeFormat(locale, { numeric: 'auto', style: 'narrow' })
```

Reply:
Good call

Reply:
done